### PR TITLE
Improve slurm service configuration

### PR DESCRIPTION
**Pull request by lsix**
In the current way `slurm` is configured, I cannot run user-space commands (such as `sinfo`) without having to specify where the slurm conf file is (it is not enough to specify it for the services).

I see 3 options so solve this:
— Always have slum read its configuration in `/etc/slurm.conf` (and maintain the config file with `environment.etc."slurm.conf"` I guess)
— Have the default environment define `SLURM_CONF` as a system-wide config (the only place I can think of is `/etc/profile` but I do not see a conviniant way to do it since it is produced in a `bash` related derivation)
— Generate in the store a folder containing the configuration and use as `sysconfigdir` when building `slurm` (but a new configuration implies a now folder to store it, and therefore a new derivation instanciation for slurm it-self)

In this PR, I opted for the 3rd option, but it will necessarily induce compilation for all final users when they will try to configure slurm. If another option should be preferred, please let me know.

### CODE REVIEW

**Comment by jagajaga** [comment]
I think it's better to do trick like here https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/x11/display-managers/slim.nix#L130. It will be like yours #2.

**Comment by copumpkin** [comment]
@lancelotsix if you need slurm utilities to know where the conf file is, you can use the existing wrapper machinery to wrap the individual utilities and set an environment variable for them

**Comment by lancelotsix** [comment]
Thanks for the feedback. I'll change that ASAP.

**Comment by lancelotsix** [comment]
I pushed a wrapper-based update.

File: nixos/modules/services/computing/slurm/slurm.nix [hunk 1]
```diff
@@ -91,38 +100,69 @@ in
 
   ###### implementation
 
-  config = mkIf (cfg.client.enable || cfg.server.enable) {
+  config =
+    let
+      wrappedSlurm = pkgs.stdenv.mkDerivation {
```
Comment (jagajaga) [inline nixos/modules/services/computing/slurm/slurm.nix]: Users will be able to run `srun`?

Comment (lancelotsix) [inline nixos/modules/services/computing/slurm/slurm.nix]: Yes, if you look [here](https://github.com/lancelotsix/nixpkgs/commit/fe0487e540f150efd97193bf3f07664c1f43b02e#diff-cb050614c6a38ed6167757722e5ee254R133), you see that `cfg.package` is not added to `environment.systemPackages`, only `wrappedSlurm` is. So unless the administrator adds `pkgs.slurm-llnl` to the system (and then generates conflicts), `srun` will be available via the wrapper script.

The [test](https://github.com/lancelotsix/nixpkgs/commit/fe0487e540f150efd97193bf3f07664c1f43b02e#diff-a444156140a24443d8679342134312a4R77) should show that it works (in this case `root` runs `srun`, I can improve it to have a regular user run `srun` if you prefer).

**Comment by jagajaga** [comment]
:+1: 

Please rename commit to `slurm service: improve config, add tests`

**Comment by lancelotsix** [comment]
I separated it in 2 commits. One for the configuration update and one for the tests additions.

**Pull request merged**